controllers/errors.controller.js

```diff
@@ -1,15 +1,14 @@
-exports.badDataType = (err, req, res, next) => {
-	if (err.code === "22P02") res.status(400).send({ msg: "Bad Request" });
-	else next(err);
-};
+exports.badDataType = (err, req, res, next) =>
+	err.code === "22P02" || err.code === "23502"
+		? res.status(400).send({ msg: "Bad Request" })
+		: next(err);
 
 exports.customErrors = (err, req, res, next) => {
 	const { status, msg } = err;
-	if (status && msg) {
-		res.status(status).send({ msg });
-	} else next(err);
+	if (status && msg) res.status(status).send({ msg });
+	else next(err);
 };
 
-exports.notFound = (req, res) => {
-	res.status(404).send({ msg: "Not found" });
-};
+exports.lastResort = (err, req, res, next) => res.status(500).send(err);
```

Comment:
Would recommend a console.log in your "last resort" error handler here - if something ever gets this far it means you are yet to encounter it and could do with seeing what's going on